tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/virutal/VirtualThreadServlet.java

```diff
@@ -0,0 +1,133 @@
+
+import ee.jakarta.tck.concurrent.framework.TestConstants;
+import ee.jakarta.tck.concurrent.framework.TestServlet;
+import jakarta.enterprise.concurrent.ManagedThreadFactory;
+import jakarta.enterprise.concurrent.ManagedThreadFactoryDefinition;
+import jakarta.servlet.annotation.WebServlet;
+
+@ManagedThreadFactoryDefinition(name = "java:app/concurrent/ThreadFactoryAnnoPlatform", virtual = false)
+@ManagedThreadFactoryDefinition(name = "java:app/concurrent/ThreadFactoryAnnoVirtual", virtual = true)
+@WebServlet("/VirtualThreadServlet")
+public class VirtualThreadServlet extends TestServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Runnable NOOP_RUNNABLE = () -> {
+    };
+
+    // TODO add virtual tests for ManagedExecutor and ManagedScheduledExcecutor
+
+    public void testPlatformThreadFactory() throws Exception {
+        ManagedThreadFactory platfromThreadFactoryAnno = InitialContext
+                .doLookup("java:app/concurrent/ThreadFactoryAnnoPlatform");
+        ManagedThreadFactory platfromThreadFactoryDD = InitialContext
+                .doLookup("java:app/concurrent/ThreadFactoryDDPlatform");
+
+        assertFalse(isVirtual(platfromThreadFactoryAnno.newThread(NOOP_RUNNABLE)),
+                "Thread Factory should not have returned a virtual thread when defined with virtual=false");
+        assertFalse(isVirtual(platfromThreadFactoryDD.newThread(NOOP_RUNNABLE)),
+                "Thread Factory should not have returned a virtual thread when defined with virtual=false");
+    }
+
+    public void testVirtualThreadFactory() throws Exception {
+        ManagedThreadFactory virtualThreadFactoryAnno = InitialContext
+                .doLookup("java:app/concurrent/ThreadFactoryAnnoVirtual");
+        ManagedThreadFactory virtualThreadFactoryDD = InitialContext
+                .doLookup("java:app/concurrent/ThreadFactoryDDVirtual");
+
+        // If assumeTrue checks pass it means the platform supports virtual threads and
+        // we can continue testing,
+        // otherwise the test will return successfully at this point and not continue.
+        assumeTrue(isVirtual(virtualThreadFactoryAnno.newThread(NOOP_RUNNABLE)));
+        assumeTrue(isVirtual(virtualThreadFactoryDD.newThread(NOOP_RUNNABLE)));
+
+        LinkedBlockingQueue<Object> results = new LinkedBlockingQueue<>();
+        Object result;
+
+        // Test virtual thread from Annotation
+        Thread annoThread = virtualThreadFactoryAnno
+                .newThread(new LookupAction(results, "java:app/concurrent/ThreadFactoryAnnoVirtual"));
+        annoThread.start();
+        result = results.poll(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        assertNotNull(result);
+        if (result instanceof Throwable)
+            throw new AssertionError("An error occured on thread.", (Throwable) result);
+
+        // Ensure queue is empty before starting next test
+        results.clear();
```

Comment:
It's safer to use a different queue so it has no possibility of being contaminated by prior usage,
```suggestion
        results = new LinkedBlockingQueue<>();
```